### ISSUE TITLE
Failed to provision volume with StorageClass "cinder-1": No suitable endpoint could be found in the service catalog.

### ISSUE THREAD

**Issue by pennpeng** [openshift/origin#15979]
[provide a description of the issue]

##### Version
[provide output of the `openshift version` or `oc version` command]
```
[root@master ~]# oc version
oc v3.6.0+c4dd4cf
kubernetes v1.6.1+5115d708d7
features: Basic-Auth GSSAPI Kerberos SPNEGO

Server https://master:8443
openshift v3.6.0+c4dd4cf
kubernetes v1.6.1+5115d708d7
```
##### Steps To Reproduce
use openstack cinder storageclass,i have test pv use cinder and is ok,but storageclass is failed.
```
kind: StorageClass
apiVersion: storage.k8s.io/v1
metadata:
  name: cinder-1
  annotations:
    storageclass.kubernetes.io/is-default-class: "true" 
provisioner: kubernetes.io/cinder
parameters:
  availability: nova 
```
```
apiVersion: v1
kind: PersistentVolumeClaim
metadata:
 name: pvc-registry
spec:
 accessModes:
  - ReadWriteOnce
 resources:
   requests:
     storage: 20Gi
```

##### Current Result
Failed to provision volume with StorageClass "cinder-1": No suitable endpoint could be found in the service catalog.
```
[root@penn-1 home]# oc describe pvc pvc-registry 
Name:		pvc-registry
Namespace:	default
StorageClass:	cinder-1
Status:		Pending
Volume:		
Labels:		<none>
Annotations:	volume.beta.kubernetes.io/storage-provisioner=kubernetes.io/cinder
Capacity:	
Access Modes:	
Events:
  FirstSeen	LastSeen	Count	From				SubObjectPath	Type		Reason			Message
  ---------	--------	-----	----				-------------	--------	------			-------
  16s		9s		2	persistentvolume-controller			Warning		ProvisioningFailed	Failed to provision volume with StorageClass "cinder-1": No suitable endpoint could be found in the service catalog.

```
##### Additional Information
[try to run `$ oc adm diagnostics` (or `oadm diagnostics`) command if possible]
[if you are reporting issue related to builds, provide build logs with `BUILD_LOGLEVEL=5`]
[consider attaching output of the `$ oc get all -o json -n <namespace>` command to the issue]
[visit https://docs.openshift.org/latest/welcome/index.html]

**Comment by pennpeng** [comment]
The log:
```
[root@penn-1 home]# oc describe pvc  pvc-registry 
Name:		pvc-registry
Namespace:	default
StorageClass:	cinder-1
Status:		Pending
Volume:		
Labels:		<none>
Annotations:	volume.beta.kubernetes.io/storage-provisioner=kubernetes.io/cinder
Capacity:	
Access Modes:	
Events:
  FirstSeen	LastSeen	Count	From				SubObjectPath	Type		Reason			Message
  ---------	--------	-----	----				-------------	--------	------			-------
  15s		10s		2	persistentvolume-controller			Warning		ProvisioningFailed	Failed to provision volume with StorageClass "cinder-1": No suitable endpoint could be found in the service catalog.
[root@penn-1 home]# tailf /var/log/messages
Aug 28 02:08:36 penn-1 dockerd-current: time="2017-08-28T02:08:36.241542328Z" level=info msg="{Action=exec, LoginUID=4294967295, PID=45899}"
Aug 28 02:08:36 penn-1 dockerd-current: time="2017-08-28T02:08:36.242480891Z" level=info msg="{Action=start, LoginUID=4294967295, PID=45899}"
Aug 28 02:08:36 penn-1 journal: {"type":"response","@timestamp":"2017-08-28T02:08:36Z","tags":[],"pid":7,"method":"head","statusCode":200,"req":{"url":"/","method":"head","headers":{"user-agent":"curl/7.29.0","host":"localhost:5601","accept":"*/*"},"remoteAddress":"127.0.0.1","userAgent":"127.0.0.1"},"res":{"statusCode":200,"responseTime":3,"contentLength":9},"message":"HEAD / 200 3ms - 9.0B"}
Aug 28 02:08:38 penn-1 origin-master: I0828 02:08:38.903330   71500 rest.go:324] Starting watch for /apis/image.openshift.io/v1/imagestreams, rv=3349 labels= fields= timeout=9m37s
Aug 28 02:08:41 penn-1 dockerd-current: time="2017-08-28T02:08:41.241925675Z" level=info msg="{Action=exec, LoginUID=4294967295, PID=45899}"
Aug 28 02:08:41 penn-1 dockerd-current: time="2017-08-28T02:08:41.244172831Z" level=info msg="{Action=start, LoginUID=4294967295, PID=45899}"
Aug 28 02:08:41 penn-1 journal: {"type":"response","@timestamp":"2017-08-28T02:08:41Z","tags":[],"pid":7,"method":"head","statusCode":200,"req":{"url":"/","method":"head","headers":{"user-agent":"curl/7.29.0","host":"localhost:5601","accept":"*/*"},"remoteAddress":"127.0.0.1","userAgent":"127.0.0.1"},"res":{"statusCode":200,"responseTime":1,"contentLength":9},"message":"HEAD / 200 1ms - 9.0B"}
Aug 28 02:08:42 penn-1 origin-node: W0828 02:08:42.755498   45899 openstack_instances.go:48] Failed to find compute endpoint: No suitable endpoint could be found in the service catalog.
Aug 28 02:08:42 penn-1 origin-node: W0828 02:08:42.755546   45899 kubelet_node_status.go:934] Failed to set some node status fields: failed to get instances from cloud provider
Aug 28 02:08:43 penn-1 origin-master: I0828 02:08:43.422231   71500 rest.go:324] Starting watch for /api/v1/resourcequotas, rv=3257 labels= fields= timeout=6m41s
Aug 28 02:08:45 penn-1 origin-node: I0828 02:08:45.173558   45899 operation_generator.go:609] MountVolume.SetUp succeeded for volume "kubernetes.io/configmap/b9705579-889d-11e7-b5e9-fa163e090ee3-config" (spec.Name: "config") pod "b9705579-889d-11e7-b5e9-fa163e090ee3" (UID: "b9705579-889d-11e7-b5e9-fa163e090ee3").
Aug 28 02:08:45 penn-1 origin-node: I0828 02:08:45.173963   45899 operation_generator.go:609] MountVolume.SetUp succeeded for volume "kubernetes.io/secret/b9705579-889d-11e7-b5e9-fa163e090ee3-certs" (spec.Name: "certs") pod "b9705579-889d-11e7-b5e9-fa163e090ee3" (UID: "b9705579-889d-11e7-b5e9-fa163e090ee3").
Aug 28 02:08:45 penn-1 origin-node: I0828 02:08:45.174404   45899 operation_generator.go:609] MountVolume.SetUp succeeded for volume "kubernetes.io/secret/b9705579-889d-11e7-b5e9-fa163e090ee3-aggregated-logging-curator-token-bdsr2" (spec.Name: "aggregated-logging-curator-token-bdsr2") pod "b9705579-889d-11e7-b5e9-fa163e090ee3" (UID: "b9705579-889d-11e7-b5e9-fa163e090ee3").
Aug 28 02:08:46 penn-1 dockerd-current: time="2017-08-28T02:08:46.240829109Z" level=info msg="{Action=exec, LoginUID=4294967295, PID=45899}"
Aug 28 02:08:46 penn-1 dockerd-current: time="2017-08-28T02:08:46.242401510Z" level=info msg="{Action=start, LoginUID=4294967295, PID=45899}"
Aug 28 02:08:46 penn-1 journal: {"type":"response","@timestamp":"2017-08-28T02:08:46Z","tags":[],"pid":7,"method":"head","statusCode":200,"req":{"url":"/","method":"head","headers":{"user-agent":"curl/7.29.0","host":"localhost:5601","accept":"*/*"},"remoteAddress":"127.0.0.1","userAgent":"127.0.0.1"},"res":{"statusCode":200,"responseTime":2,"contentLength":9},"message":"HEAD / 200 2ms - 9.0B"}
Aug 28 02:08:48 penn-1 origin-node: I0828 02:08:48.178069   45899 operation_generator.go:609] MountVolume.SetUp succeeded for volume "kubernetes.io/secret/b966d61f-889d-11e7-b5e9-fa163e090ee3-default-token-xpbpp" (spec.Name: "default-token-xpbpp") pod "b966d61f-889d-11e7-b5e9-fa163e090ee3" (UID: "b966d61f-889d-11e7-b5e9-fa163e090ee3").
Aug 28 02:08:48 penn-1 origin-master: I0828 02:08:48.784874   71500 trace.go:76] Trace[1826167536]: "Get /api/v1/namespaces/default" (started: 2017-08-28 02:08:48.073277485 +0000 UTC) (total time: 711.569521ms):
Aug 28 02:08:48 penn-1 origin-master: Trace[1826167536]: [711.569521ms] [711.569521ms] END
Aug 28 02:08:48 penn-1 origin-master: I0828 02:08:48.938581   71500 rest.go:324] Starting watch for /api/v1/replicationcontrollers, rv=4285 labels= fields= timeout=7m26s
Aug 28 02:08:49 penn-1 origin-node: I0828 02:08:49.183503   45899 operation_generator.go:609] MountVolume.SetUp succeeded for volume "kubernetes.io/secret/98dff270-889d-11e7-9dc1-fa163e090ee3-aggregated-logging-fluentd-token-3n13q" (spec.Name: "aggregated-logging-fluentd-token-3n13q") pod "98dff270-889d-11e7-9dc1-fa163e090ee3" (UID: "98dff270-889d-11e7-9dc1-fa163e090ee3").
Aug 28 02:08:49 penn-1 origin-node: I0828 02:08:49.183836   45899 operation_generator.go:609] MountVolume.SetUp succeeded for volume "kubernetes.io/secret/98dff270-889d-11e7-9dc1-fa163e090ee3-certs" (spec.Name: "certs") pod "98dff270-889d-11e7-9dc1-fa163e090ee3" (UID: "98dff270-889d-11e7-9dc1-fa163e090ee3").
Aug 28 02:08:49 penn-1 origin-node: I0828 02:08:49.184244   45899 operation_generator.go:609] MountVolume.SetUp succeeded for volume "kubernetes.io/configmap/98dff270-889d-11e7-9dc1-fa163e090ee3-config" (spec.Name: "config") pod "98dff270-889d-11e7-9dc1-fa163e090ee3" (UID: "98dff270-889d-11e7-9dc1-fa163e090ee3").
Aug 28 02:08:50 penn-1 origin-master: E0828 02:08:50.758538   71500 openstack.go:618] Unable to initialize cinder client for region: RegionOne
Aug 28 02:08:50 penn-1 origin-master: E0828 02:08:50.758563   71500 openstack_volumes.go:290] Unable to initialize cinder client for region: RegionOne
Aug 28 02:08:50 penn-1 origin-master: I0828 02:08:50.758572   71500 cinder_util.go:206] Error creating cinder volume: No suitable endpoint could be found in the service catalog.
Aug 28 02:08:50 penn-1 origin-master: I0828 02:08:50.758596   71500 pv_controller.go:1312] failed to provision volume for claim "default/pvc-registry" with StorageClass "cinder-1": No suitable endpoint could be found in the service catalog.
Aug 28 02:08:51 penn-1 origin-node: I0828 02:08:51.185155   45899 operation_generator.go:609] MountVolume.SetUp succeeded for volume "kubernetes.io/secret/b970736a-889d-11e7-b5e9-fa163e090ee3-aggregated-logging-kibana-token-vpnfh" (spec.Name: "aggregated-logging-kibana-token-vpnfh") pod "b970736a-889d-11e7-b5e9-fa163e090ee3" (UID: "b970736a-889d-11e7-b5e9-fa163e090ee3").
Aug 28 02:08:51 penn-1 origin-node: I0828 02:08:51.185332   45899 operation_generator.go:609] MountVolume.SetUp succeeded for volume "kubernetes.io/secret/b970736a-889d-11e7-b5e9-fa163e090ee3-kibana" (spec.Name: "kibana") pod "b970736a-889d-11e7-b5e9-fa163e090ee3" (UID: "b970736a-889d-11e7-b5e9-fa163e090ee3").
Aug 28 02:08:51 penn-1 origin-node: I0828 02:08:51.185521   45899 operation_generator.go:609] MountVolume.SetUp succeeded for volume "kubernetes.io/secret/b970736a-889d-11e7-b5e9-fa163e090ee3-kibana-proxy" (spec.Name: "kibana-proxy") pod "b970736a-889d-11e7-b5e9-fa163e090ee3" (UID: "b970736a-889d-11e7-b5e9-fa163e090ee3").
Aug 28 02:08:51 penn-1 dockerd-current: time="2017-08-28T02:08:51.240589182Z" level=info msg="{Action=exec, LoginUID=4294967295, PID=45899}"
Aug 28 02:08:51 penn-1 dockerd-current: time="2017-08-28T02:08:51.241923833Z" level=info msg="{Action=start, LoginUID=4294967295, PID=45899}"
Aug 28 02:08:51 penn-1 journal: {"type":"response","@timestamp":"2017-08-28T02:08:51Z","tags":[],"pid":7,"method":"head","statusCode":200,"req":{"url":"/","method":"head","headers":{"user-agent":"curl/7.29.0","host":"localhost:5601","accept":"*/*"},"remoteAddress":"127.0.0.1","userAgent":"127.0.0.1"},"res":{"statusCode":200,"responseTime":1,"contentLength":9},"message":"HEAD / 200 1ms - 9.0B"}
Aug 28 02:08:52 penn-1 origin-node: W0828 02:08:52.774270   45899 openstack_instances.go:48] Failed to find compute endpoint: No suitable endpoint could be found in the service catalog.
Aug 28 02:08:52 penn-1 origin-node: W0828 02:08:52.774301   45899 kubelet_node_status.go:934] Failed to set some node status fields: failed to get instances from cloud provider
Aug 28 02:08:53 penn-1 origin-master: I0828 02:08:53.817394   71500 rest.go:324] Starting watch for /api/v1/nodes, rv=144656 labels= fields= timeout=8m18s
Aug 28 02:08:56 penn-1 dockerd-current: time="2017-08-28T02:08:56.240673057Z" level=info msg="{Action=exec, LoginUID=4294967295, PID=45899}"
Aug 28 02:08:56 penn-1 dockerd-current: time="2017-08-28T02:08:56.242261498Z" level=info msg="{Action=start, LoginUID=4294967295, PID=45899}"
Aug 28 02:08:56 penn-1 journal: {"type":"response","@timestamp":"2017-08-28T02:08:56Z","tags":[],"pid":7,"method":"head","statusCode":200,"req":{"url":"/","method":"head","headers":{"user-agent":"curl/7.29.0","host":"localhost:5601","accept":"*/*"},"remoteAddress":"127.0.0.1","userAgent":"127.0.0.1"},"res":{"statusCode":200,"responseTime":2,"contentLength":9},"message":"HEAD / 200 2ms - 9.0B"}
Aug 28 02:09:01 penn-1 dockerd-current: time="2017-08-28T02:09:01.240473837Z" level=info msg="{Action=exec, LoginUID=4294967295, PID=45899}"
Aug 28 02:09:01 penn-1 dockerd-current: time="2017-08-28T02:09:01.241462852Z" level=info msg="{Action=start, LoginUID=4294967295, PID=45899}"
Aug 28 02:09:01 penn-1 journal: {"type":"response","@timestamp":"2017-08-28T02:09:01Z","tags":[],"pid":7,"method":"head","statusCode":200,"req":{"url":"/","method":"head","headers":{"user-agent":"curl/7.29.0","host":"localhost:5601","accept":"*/*"},"remoteAddress":"127.0.0.1","userAgent":"127.0.0.1"},"res":{"statusCode":200,"responseTime":1,"contentLength":9},"message":"HEAD / 200 1ms - 9.0B"}
Aug 28 02:09:02 penn-1 origin-node: W0828 02:09:02.786086   45899 openstack_instances.go:48] Failed to find compute endpoint: No suitable endpoint could be found in the service catalog.
Aug 28 02:09:02 penn-1 origin-node: W0828 02:09:02.786116   45899 kubelet_node_status.go:934] Failed to set some node status fields: failed to get instances from cloud provider
Aug 28 02:09:02 penn-1 origin-master: I0828 02:09:02.889630   71500 rest.go:324] Starting watch for /apis/extensions/v1beta1/replicasets, rv=3257 labels= fields= timeout=6m36s
Aug 28 02:09:05 penn-1 origin-master: E0828 02:09:05.749363   71500 openstack.go:618] Unable to initialize cinder client for region: RegionOne
Aug 28 02:09:05 penn-1 origin-master: E0828 02:09:05.749381   71500 openstack_volumes.go:290] Unable to initialize cinder client for region: RegionOne
Aug 28 02:09:05 penn-1 origin-master: I0828 02:09:05.749388   71500 cinder_util.go:206] Error creating cinder volume: No suitable endpoint could be found in the service catalog.
Aug 28 02:09:05 penn-1 origin-master: I0828 02:09:05.749405   71500 pv_controller.go:1312] failed to provision volume for claim "default/pvc-registry" with StorageClass "cinder-1": No suitable endpoint could be found in the service catalog.
Aug 28 02:09:06 penn-1 dockerd-current: time="2017-08-28T02:09:06.241115314Z" level=info msg="{Action=exec, LoginUID=4294967295, PID=45899}"
Aug 28 02:09:06 penn-1 dockerd-current: time="2017-08-28T02:09:06.241912326Z" level=info msg="{Action=start, LoginUID=4294967295, PID=45899}"
Aug 28 02:09:06 penn-1 journal: {"type":"response","@timestamp":"2017-08-28T02:09:06Z","tags":[],"pid":7,"method":"head","statusCode":200,"req":{"url":"/","method":"head","headers":{"user-agent":"curl/7.29.0","host":"localhost:5601","accept":"*/*"},"remoteAddress":"127.0.0.1","userAgent":"127.0.0.1"},"res":{"statusCode":200,"responseTime":1,"contentLength":9},"message":"HEAD / 200 1ms - 9.0B"}
Aug 28 02:09:09 penn-1 origin-master: W0828 02:09:09.587879   71500 openstack_instances.go:48] Failed to find compute endpoint: No suitable endpoint could be found in the service catalog.
Aug 28 02:09:09 penn-1 origin-master: E0828 02:09:09.587989   71500 operation_generator.go:163] VolumesAreAttached failed for checking on node "penn-2" with: failed to list openstack instances
Aug 28 02:09:11 penn-1 dockerd-current: time="2017-08-28T02:09:11.241446731Z" level=info msg="{Action=exec, LoginUID=4294967295, PID=45899}"
Aug 28 02:09:11 penn-1 dockerd-current: time="2017-08-28T02:09:11.242061118Z" level=info msg="{Action=start, LoginUID=4294967295, PID=45899}"
Aug 28 02:09:11 penn-1 journal: {"type":"response","@timestamp":"2017-08-28T02:09:11Z","tags":[],"pid":7,"method":"head","statusCode":200,"req":{"url":"/","method":"head","headers":{"user-agent":"curl/7.29.0","host":"localhost:5601","accept":"*/*"},"remoteAddress":"127.0.0.1","userAgent":"127.0.0.1"},"res":{"statusCode":200,"responseTime":1,"contentLength":9},"message":"HEAD / 200 1ms - 9.0B"}
Aug 28 02:09:11 penn-1 origin-node: I0828 02:09:11.414863   45899 qos_container_manager_linux.go:285] [ContainerManager]: Updated QoS cgroup configuration
Aug 28 02:09:12 penn-1 origin-master: I0828 02:09:12.384337   71500 rest.go:324] Starting watch for /api/v1/namespaces, rv=3257 labels= fields= timeout=6m42s
Aug 28 02:09:12 penn-1 origin-node: W0828 02:09:12.798909   45899 openstack_instances.go:48] Failed to find compute endpoint: No suitable endpoint could be found in the service catalog.
Aug 28 02:09:12 penn-1 origin-node: W0828 02:09:12.798934   45899 kubelet_node_status.go:934] Failed to set some node status fields: failed to get instances from cloud provider
Aug 28 02:09:12 penn-1 origin-master: I0828 02:09:12.881915   71500 rest.go:324] Starting watch for /api/v1/resourcequotas, rv=3257 labels= fields= timeout=7m50s
Aug 28 02:09:12 penn-1 origin-master: I0828 02:09:12.901638   71500 rest.go:324] Starting watch for /apis/authorization.openshift.io/v1/policies, rv=3257 labels= fields= timeout=8m57s
Aug 28 02:09:16 penn-1 dockerd-current: time="2017-08-28T02:09:16.241258352Z" level=info msg="{Action=exec, LoginUID=4294967295, PID=45899}"
Aug 28 02:09:16 penn-1 dockerd-current: time="2017-08-28T02:09:16.242596669Z" level=info msg="{Action=start, LoginUID=4294967295, PID=45899}"
Aug 28 02:09:16 penn-1 journal: {"type":"response","@timestamp":"2017-08-28T02:09:16Z","tags":[],"pid":7,"method":"head","statusCode":200,"req":{"url":"/","method":"head","headers":{"user-agent":"curl/7.29.0","host":"localhost:5601","accept":"*/*"},"remoteAddress":"127.0.0.1","userAgent":"127.0.0.1"},"res":{"statusCode":200,"responseTime":2,"contentLength":9},"message":"HEAD / 200 2ms - 9.0B"}
^C

```

**Comment by pennpeng** [comment]
when i restart the system or restart the origin service it  will be ok for a moment,after a little time ,Again, there's the problem.

**Comment by pennpeng** [comment]
and i create a svc use loadblance ，cant't work for openstack loadbance:
```
Error creating load balancer (will retry): Failed to create load balancer for service default/nginx-svc: Error creating loadbalancer a423a38568bbf11e7b278fa163e090ee: Error creating loadbalancer {a423a38568bbf11e7b278fa163e090ee Kubernetes external service a423a38568bbf11e7b278fa163e090ee 27da7c46-db27-490a-9d38-2fa1b3b126c9 <nil> }: Resource not found
```

**Comment by tourunen** [comment]
We saw the same problem when deploying Origin 3.6 with openshift-ansible. It is likely related to these issues:

https://github.com/gophercloud/gophercloud/issues/255
https://github.com/kubernetes/kubernetes/issues/44461

We were using Keystone API version v3 successfully with Origin 1.5, but with 3.6 the Cinder driver started freezing. Based on brief testing, changing the authentication URL endpoint to v2.0 seems to be a workaround.

The configuration guide example [1] uses v2.0 endpoint, maybe that is why not many are hitting this.

[1] https://docs.openshift.org/latest/install_config/configuring_openstack.html#openstack-configuring-masters-ansible

**Comment by pospispa** [comment]
@pennpeng will you, please, share your `cloud.conf` file the `[BlockStorage]` section, `bs-version` parameter. Will you also share your OpenStack service catalog, the Cinder entries.

**Comment by pennpeng** [comment]
@pospispa  hi,
```
[root@penn-1 cloudprovider]# ls
openstack.conf
[root@penn-1 cloudprovider]# cat openstack.conf 
[Global]
auth-url = http://172.18.22.100:5000/v3
username = caowei
password = caowei
domain-id = default
tenant-id = 40844d061473409daebe3f4cad2880a6
region = RegionOne
[LoadBalancer]
subnet-id = c95a92d1-fa5a-4b1e-b8a6-dc59ef254641
[root@penn-1 cloudprovider]# pwd
/etc/origin/cloudprovider
```
openstack version ocata

**Comment by pospispa** [comment]
@pennpeng the error you see is reported [here](https://github.com/openshift/origin/blob/c4dd4cf368b6204571faacde702e624b930a5214/vendor/github.com/gophercloud/gophercloud/openstack/endpoint_location.go#L97).

OpenShift authenticates to OpenStack Keystone and receives service catalog with a list of all available services.

I assume that OpenShift is trying to create Cinder V1 client (default behaviour as `bs-version` is not configured) so I assume that the [`V3EndpointURL` func](https://github.com/openshift/origin/blob/c4dd4cf368b6204571faacde702e624b930a5214/vendor/github.com/gophercloud/gophercloud/openstack/endpoint_location.go#L63) is trying to find Cinder V1 service in the service catalog but couldn't find any.

So I recommend to check if Cinder V1 service is available in you OpenStack cluster if it's available to the `caowei` user and if it's in the `RegionOne` region.

You may check OpenStack logs for communication between OpenShift and OpenStack.

The search for Cinder V1 in the received service catalog on OpenShift side is done in `gophercloud` library that doesn't log any information about the service catalog parsing and the search for Cinder V1 service.

Note: OpenShift supports only Cinder V1 and Cinder V2.

**Comment by pospispa** [comment]
I've re-read the description of the issue and the related upstream issues:
- https://github.com/gophercloud/gophercloud/issues/255
- https://github.com/kubernetes/kubernetes/issues/44461
and I believe that this issue is the same as the upstream issues.

As the upstream issues are fixed in Kubernetes 1.7 this issue will be fixed in OpenShift 3.7.

**Comment by briantward** [comment]
This was fixed in OpenShift 3.7 and backported to OpenShift 3.6
https://bugzilla.redhat.com/show_bug.cgi?id=1491331
https://bugzilla.redhat.com/show_bug.cgi?id=1527973
